axlearn/common/trainer.py

```diff
@@ -556,6 +556,11 @@ def run(
                 output = None
                 stop_trace_step = None
 
+                # skip previous batches
+                for i in range(self.step):
+                    self.vlog(3, f"skipping batch for step {i}")
```

Comment:
Do we need to add any unit tests for this change?

Reply:
No unit tests planed for now since these are not merging to upstream.